lib/Basic/Subprocess.cpp

```diff
@@ -436,17 +457,44 @@ void llbuild::basic::spawnProcess(
 
     // If we have been cancelled since we started, do nothing.
     if (!wasCancelled) {
-      int result = posix_spawn(&pid, args[0], /*file_actions=*/&fileActions,
-                               /*attrp=*/&attributes, const_cast<char**>(args.data()),
-                               const_cast<char* const*>(environment.getEnvp()));
+      int result = 0;
+
+#ifdef __APPLE__
+      thread_local std::string threadWorkingDir;
```

Comment:
I'm not sure we should use thread_local, would it be better to store this in the LaneBasedExecutionQueueJobContext?

Reply:
We could store it in the ProcessContext, but right now that is an opaque type that is merely passed through the process code and back to the queue.  In the lane queue, that context is created per task, thus doesn't stay resident for the lifetime of the thread.   Both can be changed, of course, but it makes using 'spawnProcess' a bit more fragile.   I think the overhead of the thread_local is minimal in this case, given that it is function scoped.

Reply:
Ok, works for me!